api/repositories/service_offering_repository.go

```diff
@@ -61,6 +62,30 @@ func NewServiceOfferingRepo(
 	}
 }
 
+func (r *ServiceOfferingRepo) GetServiceOffering(ctx context.Context, authInfo authorization.Info, guid string) (ServiceOfferingRecord, error) {
+	userClient, err := r.userClientFactory.BuildClient(authInfo)
+	if err != nil {
+		return ServiceOfferingRecord{}, fmt.Errorf("failed to build user client: %w", err)
+	}
+
+	offering := &korifiv1alpha1.CFServiceOffering{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: r.rootNamespace,
+			Name:      guid,
+		},
+	}
+
+	if err = userClient.Get(ctx, client.ObjectKeyFromObject(offering), offering); err != nil {
+		if k8serrors.IsForbidden(err) {
+			return ServiceOfferingRecord{}, nil
```

Comment:
We should not be ignoring the forbidden error on `Get`. Instead, we should translate it to korifi forbidden error. You have actually done that below.
Lets just get rid of the forbidden check.

Reply:
Done